src/utils/contract.ts

```diff
@@ -119,4 +120,37 @@ export default class Contract extends BaseConnection {
     });
     return await response.json();
   }
+
+  private async fetchDataViaProvider(params) {
+    switch (this.providerType) {
+      case "ethers":
+        return await this.web3Provider!
+          .sendAsync('eth_call', params)
+          .then(resp => resp.json()); 
+      case "web3":
+        try {
+          return this.web3Provider?.sendAsync({
```

Comment:
If the implementation doesn't return a `Promise`, this call won't return anything at all. But since we provide a callback, we can promisify this call to make sure there's always a promise (the same applies to `send`)